services/web-app/utils/use-query-state.js

```diff
@@ -57,15 +28,35 @@ export const useQueryState = (
       return null
     }
 
-    const query = new URLSearchParams(window.location.search)
-    const queryValue = query.get(key)
-    const storageValue = saveToStorage ? localStorage.getItem(`${router.pathname}:${key}`) : null
+    const query = queryString.parseUrl(window.location.search, {
+      ...queryStringConfig,
+      parseNumbers,
+      parseBooleans
+    }).query
+
+    const storageValue = saveToStorage
+      ? localStorage.getItem(`${router.pathname}:${key}`)
+      : undefined
+
+    let val = query[key]
 
-    return queryValue !== null ? parse(queryValue) : storageValue
-  }, [key, parse, router.pathname, saveToStorage])
+    if (val !== undefined && validateValue && !validateValue(val)) {
```

Comment:
super extra safety here would be to do something like:
```suggestion
    if (val !== undefined && typeof validateValue === 'function' && !validateValue(val)) {
```